### PR TITLE
Update eslint plugins

### PR DESCRIPTION
**Pull request by amanmahajan7**
https://github.com/typescript-eslint/typescript-eslint/blob/main/CHANGELOG.md
https://github.com/jest-community/eslint-plugin-jest/blob/main/CHANGELOG.md
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/CHANGELOG.md

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -629,31 +631,18 @@ const rules = {
...
   '@typescript-eslint/switch-exhaustiveness-check': 1,
   '@typescript-eslint/triple-slash-reference': [1, { path: 'never', types: 'never', lib: 'never' }],
-  '@typescript-eslint/type-annotation-spacing': 0,
   '@typescript-eslint/typedef': 0,
   '@typescript-eslint/unbound-method': 0,
   '@typescript-eslint/unified-signatures': 0,
 
   // @typescript-eslint/eslint-plugin Extension Rules
   // https://typescript-eslint.io/rules/#extension-rules
-  '@typescript-eslint/block-spacing': 0,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:639]: https://typescript-eslint.io/blog/deprecating-formatting-rules/#upgrading-to-eslint-stylistic

File: .eslintrc.cjs [hunk 2]
```diff
@@ -614,13 +616,13 @@ const rules = {
...
   ],
   '@typescript-eslint/prefer-optional-chain': 1,
+  '@typescript-eslint/prefer-promise-reject-errors': [1, { allowEmptyReject: true }],
   '@typescript-eslint/prefer-readonly': 1,
   '@typescript-eslint/prefer-readonly-parameter-types': 0,
   '@typescript-eslint/prefer-reduce-type-parameter': 1,
   '@typescript-eslint/prefer-regexp-exec': 1,
   '@typescript-eslint/prefer-return-this-type': 0,
   '@typescript-eslint/prefer-string-starts-ends-with': 1,
-  '@typescript-eslint/prefer-ts-expect-error': 1,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:623]: Replaced by https://typescript-eslint.io/rules/ban-tslint-comment/

File: .eslintrc.cjs [hunk 3]
```diff
@@ -554,17 +552,18 @@ const rules = {
   '@typescript-eslint/explicit-function-return-type': 0,
   '@typescript-eslint/explicit-member-accessibility': 0,
   '@typescript-eslint/explicit-module-boundary-types': 0,
-  '@typescript-eslint/member-delimiter-style': 0,
   '@typescript-eslint/member-ordering': 0,
   '@typescript-eslint/method-signature-style': 1,
   '@typescript-eslint/naming-convention': 0,
+  '@typescript-eslint/no-array-delete': 1,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:558]: https://typescript-eslint.io/rules/no-array-delete/

File: .eslintrc.cjs [hunk 4]
```diff
@@ -554,17 +552,18 @@ const rules = {
...
   '@typescript-eslint/no-base-to-string': 0,
   '@typescript-eslint/no-confusing-non-null-assertion': 0,
   '@typescript-eslint/no-confusing-void-expression': [1, { ignoreArrowShorthand: true }],
   '@typescript-eslint/no-duplicate-enum-values': 1,
   '@typescript-eslint/no-duplicate-type-constituents': 1,
   '@typescript-eslint/no-dynamic-delete': 0,
   '@typescript-eslint/no-empty-interface': 1,
+  '@typescript-eslint/no-empty-object-type': 1,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:566]: https://typescript-eslint.io/rules/no-empty-object-type/

File: .eslintrc.cjs [hunk 5]
```diff
@@ -587,7 +586,9 @@ const rules = {
   '@typescript-eslint/no-type-alias': 0,
   '@typescript-eslint/no-unnecessary-boolean-literal-compare': 1,
   '@typescript-eslint/no-unnecessary-condition': 1,
+  '@typescript-eslint/no-unnecessary-parameter-property-assignment': 1,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:589]: https://typescript-eslint.io/rules/no-unnecessary-parameter-property-assignment/

File: .eslintrc.cjs [hunk 6]
```diff
@@ -587,7 +586,9 @@ const rules = {
...
   '@typescript-eslint/no-unnecessary-qualifier': 0,
+  '@typescript-eslint/no-unnecessary-template-expression': 1,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:591]: https://typescript-eslint.io/rules/no-unnecessary-template-expression/

File: .eslintrc.cjs [hunk 7]
```diff
@@ -604,6 +605,7 @@ const rules = {
   '@typescript-eslint/parameter-properties': 1,
   '@typescript-eslint/prefer-as-const': 1,
   '@typescript-eslint/prefer-enum-initializers': 0,
+  '@typescript-eslint/prefer-find': 1,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:608]: https://typescript-eslint.io/rules/prefer-find/

File: .eslintrc.cjs [hunk 8]
```diff
@@ -709,13 +699,9 @@ const rules = {
...
     }
   ],
-  '@typescript-eslint/quotes': 0,
   '@typescript-eslint/require-await': 1,
   '@typescript-eslint/return-await': 1,
-  '@typescript-eslint/semi': 0,
-  '@typescript-eslint/space-before-blocks': 0,
-  '@typescript-eslint/space-before-function-paren': 0,
-  '@typescript-eslint/space-infix-ops': 0,
+  '@typescript-eslint/use-unknown-in-catch-callback-variable': 1,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:704]: https://typescript-eslint.io/rules/use-unknown-in-catch-callback-variable/

File: .eslintrc.cjs [hunk 9]
```diff
@@ -432,6 +432,7 @@ const rules = {
   'jest/no-commented-out-tests': 1,
   'jest/no-conditional-expect': 1,
   'jest/no-conditional-in-test': 0,
+  'jest/no-confusing-set-timeout': 1,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:435]: https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-confusing-set-timeout.md

File: .eslintrc.cjs [hunk 10]
```diff
@@ -463,9 +464,11 @@ const rules = {
   'jest/prefer-each': 1,
   'jest/prefer-equality-matcher': 1,
   'jest/prefer-expect-assertions': 0,
-  'jest/prefer-expect-resolves': 0,
+  'jest/prefer-expect-resolves': 1,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:467]: I think we can enable it. Wdyt?
https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/prefer-expect-resolves.md

File: .eslintrc.cjs [hunk 11]
```diff
@@ -287,6 +287,7 @@ const rules = {
   // https://github.com/jsx-eslint/eslint-plugin-react#list-of-supported-rules
   'react/boolean-prop-naming': 0,
   'react/button-has-type': 0,
+  'react/checked-requires-onchange-or-readonly': 1,
```
Comment (amanmahajan7) [inline .eslintrc.cjs:290]: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/checked-requires-onchange-or-readonly.md

File: .eslintrc.cjs [hunk 12]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (nstepien) [inline .eslintrc.cjs:1]: Should keep this

File: .eslintrc.cjs (hunk 9 above)
Comment (nstepien) [reply .eslintrc.cjs:435]: Does vitest have an equivalent API?
https://vitest.dev/api/vi.html

File: .eslintrc.cjs (hunk 2 above)
Comment (nstepien) [reply .eslintrc.cjs:623]: >This rule has been deprecated in favor of @typescript-eslint/ban-ts-comment.

Confused `ban-tslint-comment` and `ban-ts-comment`?

We don't use tslint, so maybe we can keep `ban-tslint-comment` disabled?

File: .eslintrc.cjs (hunk 8 above)
Comment (nstepien) [reply .eslintrc.cjs:707]: Should we enable this as well?
https://www.typescriptlang.org/tsconfig/#useUnknownInCatchVariables

File: test/column/frozen.test.ts [hunk 13]
```diff
@@ -30,8 +30,8 @@ test('frozen column have a specific class, and are stable-sorted before non-froz
 
   expect(cell1).toHaveClass(`${cellClassname} ${cellFrozenClassname}`, { exact: true });
   expect(cell2).toHaveClass(`${cellClassname} ${cellFrozenClassname}`, { exact: true });
-  expect(cell3).toHaveClass(`${cellClassname}`, { exact: true });
-  expect(cell4).toHaveClass(`${cellClassname}`, { exact: true });
```
Comment (nstepien) [inline test/column/frozen.test.ts:34]: Was this caught by a new rule?

Comment (amanmahajan7) [reply test/column/frozen.test.ts:34]: Yeah

File: .eslintrc.cjs (hunk 12 above)
Comment (amanmahajan7) [reply .eslintrc.cjs:1]: Biome removes it for some reason

File: .eslintrc.cjs (hunk 9 above)
Comment (amanmahajan7) [reply .eslintrc.cjs:437]: https://vitest.dev/guide/migration#timeout 
Should I disable it?

Comment (nstepien) [reply .eslintrc.cjs:437]: Let's disable it, every enabled rule takes time to process

File: .eslintrc.cjs [hunk 14]
```diff
@@ -201,7 +201,7 @@ const rules = {
   'prefer-numeric-literals': 1,
   'prefer-object-has-own': 1,
   'prefer-object-spread': 1,
-  'prefer-promise-reject-errors': [1, { allowEmptyReject: true }],
+  'prefer-promise-reject-errors': 'off', // replaced by @typescript-eslint/prefer-promise-reject-errors
```
Comment (nstepien) [inline .eslintrc.cjs:204]:
```suggestion
  'prefer-promise-reject-errors': 0, // replaced by @typescript-eslint/prefer-promise-reject-errors
```

**Review by nstepien — APPROVED** [review]

**Review by nstepien — APPROVED** [review]

**Pull request merged**